snoop/data/management/commands/filestats.py

```diff
@@ -0,0 +1,181 @@
+from django.db.models.expressions import RawSQL
+
+from ...analyzers import archives
+from ...analyzers import tika
+from ...analyzers import email
+from ...analyzers import exif
+from ...analyzers import html
+from ... import filesystem
+
+
+SUPPORTED_MIME_TYPES = (archives.ARCHIVES_MIME_TYPES
+                        .union(tika.TIKA_MIME_TYPES)
+                        .union(filesystem.EMLX_EMAIL_MIME_TYPES)
+                        .union(email.OUTLOOK_POSSIBLE_MIME_TYPES)
+                        .union(filesystem.RFC822_EMAIL_MIME_TYPES)
+                        .union(exif.EXIFREAD_MIME_TYPES)
+                        .union(html.HTML_MIME_TYPES))
+
+
+def truncate_size(size):
+    """Generate a truncated number for a given number.
+
+    This is needed to anonymize the statistics, so they can't be traced back
+    to some dataset.
+    """
+    return round(size, -((len(str(size))) - 1))
+
+
+def get_top_mime_types(collections_list, print_supported=True):
+    """Return a dictionary of mime-types that occupy most space in collections.
+
+    Args:
+        collections_list: A list of collections that will be analyzed.
+        print_supported: When False only analyzes unsupported filetypes.
+    """
+    res = {}
+    for col in collections_list:
+        collection = collections.ALL[col]
+        with collection.set_current():
+            queryset_mime = models.Blob.objects.all().values('mime_type', 'magic')\
+                .annotate(total=Count('mime_type')).annotate(size=Sum('size'))\
+                .order_by('-size')
+            if not print_supported:
+                queryset_mime = queryset_mime.exclude(mime_type__in=SUPPORTED_MIME_TYPES)
+            for mtype in queryset_mime:
+                if mtype['mime_type'] not in res:
+                    res[mtype['mime_type']] = {'size': truncate_size(mtype['size']),
+                                               'magic': get_description(col, mtype['mime_type'])}
+                else:
+                    res[mtype['mime_type']]['size'] += truncate_size(mtype['size'])
+    sorted_res = sorted(res.items(), key=lambda x: x[1]['size'], reverse=True)
+    return dict(sorted_res)
+
+
+def get_top_extensions(collections_list, print_supported=True):
+    """Return a dictionary of file extensions that occupy most space in collections.
+
+    Args:
+        collections_list: A list of collections that will be analyzed.
+        print_supported: When False only analyzes unsupported filetypes.
+    """
+    ext_dict = {}
+    for col in collections_list:
+        query = r"""select substring(encode(f.name_bytes::bytea, 'escape')::text
+                    from '(\..{1,20})$') as ext,
+                    sum(f.size) as size,
+                    b.mime_type as mime
+                    from data_file f
+                    join data_blob b on f.blob_id = b.sha3_256
+                    group by ext, mime
+                    order by size desc limit 100;"""
```

Comment:
Put the 100 in a new argument `--row-count`, i want to run it on 1000 extensions too.